server/src/test/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResourceTest.java

```diff
@@ -0,0 +1,103 @@
+package keywhiz.service.resources.automation.v2;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import java.util.UUID;
+import keywhiz.api.ApiDate;
+import keywhiz.api.model.AutomationClient;
+import keywhiz.api.model.Client;
+import keywhiz.jooq.tables.records.SecretsRecord;
+import keywhiz.log.AuditLog;
+import keywhiz.log.Event;
+import keywhiz.log.EventTag;
+import keywhiz.service.crypto.RowHmacGenerator;
+import keywhiz.test.TestDSLContexts;
+import org.jooq.DSLContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class BackfillRowHmacResourceTest {
+  @Mock
+  private RowHmacGenerator rowHmacGenerator;
+  @Mock
+  private AuditLog auditLog;
+
+  @Captor
+  private ArgumentCaptor<Event> eventCaptor;
+
+  @Before
+  public void before() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void backfillSecretHmacByNameHandlesNullOldHmac() {
+    SecretsRecord record = new SecretsRecord();
+    record.setId(1L);
+    record.setName("foo");
+    record.setRowHmac(null);
+
+    DSLContext context = TestDSLContexts.returning(record);
+
+    when(rowHmacGenerator.computeRowHmac(any(), any())).thenReturn("newHmac");
+
+    BackfillRowHmacResource resource = new BackfillRowHmacResource(context, rowHmacGenerator, auditLog);
+
+    String clientName = UUID.randomUUID().toString();
+    String secretName = UUID.randomUUID().toString();
+
+    AutomationClient automationClient = newAutomationClient(clientName);
+
+    resource.backfillSecretRowHmacByName(automationClient, secretName, true);
+  }
+
+  @Test
+  public void backfillSecretHmacByNameWritesToAuditLog() {
+    String oldHmac = UUID.randomUUID().toString();
+    String newHmac = UUID.randomUUID().toString();
+
+    when(rowHmacGenerator.computeRowHmac(any(), any())).thenReturn(newHmac);
+
+    SecretsRecord record = new SecretsRecord();
+    record.setId(1L);
+    record.setName("foo");
+    record.setRowHmac(oldHmac);
+
+    DSLContext context = TestDSLContexts.returning(record);
```

Comment:
The DSLContext's builder-ish API isn't amenable to trivial mocking but I'm happy with this simple alternative, which is basically shorthand for when-select-then-return-thing.